ui/src/App.tsx

```diff
@@ -377,19 +385,45 @@ const defaultImmunizations: Promise<StoredSHC[]> = Promise.all(
 
 export function SHLinkCreate() {
   let navigate = useNavigate();
-  let { store } = useStore();
+  let location = useLocation();
+  let { store, dispatch } = useStore();
   let [usePasscode, setUsePasscode] = useState(false);
   let [passcode, setPasscode] = useState('1234');
   let [expires, setExpires] = useState(false);
 
+  const { custom } = location.state as LocationState || {}; 
+
+  const handleOnChange = (index: Number) => {
+    const updatedCheck = isChecked.map((item: boolean, i: Number) => {
+      return i === index ? !item : item
+    });
+    setIsChecked(updatedCheck);
+  }
+
   let oneMonthExpiration = new Date(new Date().getTime() + 1000 * 3600 * 24 * 31);
   let [expiresDate, setExpiresDate] = useState(oneMonthExpiration.toISOString().slice(0, 10));
   let [searchParams] = useSearchParams();
   let datasetId = Number(searchParams.get('ds'));
   let ds = store.sharing[datasetId];
-  let vaccines = Object.values(store.vaccines).filter(filterForTypes(ds.shcTypes));
+  let vaccines = (custom ? Object.values(store.vaccines) : Object.values(store.vaccines).filter(filterForTypes(ds.shcTypes)).filter(filterForIds(ds.shcs)));
+  const defaultArray = new Array(vaccines.length).fill(false)
+  // state for keeping track of checked/unchecked vaccines
+  const [isChecked, setIsChecked] = useState<boolean[]>(defaultArray);
 
   async function activate() {
+    if (custom) {
+      const checkedVaccinations = vaccines.map(card => card.id).filter((card, i) => isChecked[i] === true);
+      // create new DataSet using the checked vaccinations
+      // TODO: Allow user to rename the dataset
+      datasetId = Object.keys(store.sharing).length;
+      const customDataSet : DataSet = {
+        id: datasetId,
+        name: `Custom DataSet ${datasetId}`,
```

Comment:
It'd be great to land this functionality before we merge -- may be as simple as just adding an `<input>` and populating its default value to `Custom Data Set ${datasetId}` but allowing the user to type their own value inside, and and just grabbing that value when `activate()` is called.